components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.addons
+
+import mozilla.components.browser.state.action.WebExtensionAction
+import mozilla.components.browser.state.state.WebExtensionState
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.support.webextensions.WebExtensionSupport
+
+/**
+ * Provides access to installed and recommended [Addon]s and manages their state.
+ *
+ * @property store The application's [BrowserStore].
+ * @property addonsProvider The [AddonsProvider] to query available [Addon]s.
+ */
+class AddonManager(
+    private val store: BrowserStore,
+    private val addonsProvider: AddonsProvider
+) {
+
+    /**
+     * Returns the list of all installed and recommended add-ons.
+     *
+     * @return list of all [Addon]s with up-to-date [Addon.installedState].
+     * @throws AddonManagerException in case of a problem reading from
+     * the [addonsProvider] or querying web extension state from the engine / store.
+     */
+    @Throws(AddonManagerException::class)
+    @Suppress("TooGenericExceptionCaught")
+    suspend fun getAddons(): List<Addon> {
```

Comment:
Should we just rename this method to `getInstalledAddons`? `getAddons` sounds too generic for me, I would have to look at the javadoc.

Reply:
The method returns all `Addon`s (incl. the recommended but not installed ones). So, I think we should keep the current name or maybe use `getAllAddons`...

Reply:
Sounds good